File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/POCOViewModelsModule.vb
Imports MVVMDemo.POCOViewModels
Imports DevExpress.Xpf.DemoBase

Namespace MVVMDemo

    Public Partial Class POCOViewModelsModule
        Inherits MVVMDemoModule

        Protected Overrides Function CreateShowcases() As IShowcaseInfo()
            Const path As String = "Modules/POCOViewModels"
            Const uriPrefix As String = "17352/Common-Concepts/MVVM-Framework/ViewModels/POCO-ViewModels#"
            Return New ShowcaseInfo() {LoadShowcase("Overview", uriPrefix & "Basics", path, {GetType(LoginViewModel), GetType(LoginView), GetType(LoginViewModel_RuntimeGeneratedCode)}), LoadShowcase("Bindable Properties", uriPrefix & "BindableProperties", path, {GetType(PropertiesViewModel), GetType(PropertiesView), GetType(PropertiesViewModel_RuntimeGeneratedCode)}), LoadShowcase("Commands", uriPrefix & "Commands", path, {GetType(CommandsViewModel), GetType(CommandsView), GetType(CommandsViewModel_RuntimeGeneratedCode)}), LoadShowcase("Services", uriPrefix & "Services", path, {GetType(ServicesViewModel), GetType(ServicesView), GetType(ServicesViewModel_RuntimeGeneratedCode)}), LoadShowcase("IDataErrorInfo", uriPrefix & "IDataErrorInfo", path, {GetType(DataErrorInfoViewModel), GetType(DataErrorInfoView), GetType(DataErrorInfoViewModel_RuntimeGeneratedCode)}), LoadShowcase("Dependent Properties", Nothing, path, {GetType(DependentPropertiesViewModel), GetType(DependentPropertiesView), GetType(DependentPropertiesViewModel_RuntimeGeneratedCode)}), LoadShowcase("Change Notifications", Nothing, path, {GetType(ChangeNotificationsViewModel), GetType(ChangeNotificationsView), GetType(ChangeNotificationsViewModel_RuntimeGeneratedCode)})}
        End Function
    End Class
End Namespace

File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/POCOViewModelsModule.cs
using MVVMDemo.POCOViewModels;
using DevExpress.Xpf.DemoBase;

namespace MVVMDemo {
    public partial class POCOViewModelsModule : MVVMDemoModule {
        protected override IShowcaseInfo[] CreateShowcases() {
            const string path = "Modules/POCOViewModels";
            const string uriPrefix = "17352/Common-Concepts/MVVM-Framework/ViewModels/POCO-ViewModels#";
            return new ShowcaseInfo[] {
                LoadShowcase("Overview", uriPrefix + "Basics", path, new[] { typeof(LoginViewModel), typeof(LoginView), typeof(LoginViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("Bindable Properties", uriPrefix + "BindableProperties", path, new[] { typeof(PropertiesViewModel), typeof(PropertiesView), typeof(PropertiesViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("Commands", uriPrefix + "Commands", path, new[] { typeof(CommandsViewModel), typeof(CommandsView), typeof(CommandsViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("Services", uriPrefix + "Services", path, new[] { typeof(ServicesViewModel), typeof(ServicesView), typeof(ServicesViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("IDataErrorInfo", uriPrefix + "IDataErrorInfo", path, new[] { typeof(DataErrorInfoViewModel), typeof(DataErrorInfoView), typeof(DataErrorInfoViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("Dependent Properties", null, path, new[] { typeof(DependentPropertiesViewModel), typeof(DependentPropertiesView), typeof(DependentPropertiesViewModel_RuntimeGeneratedCode) }),
                LoadShowcase("Change Notifications", null, path, new[] { typeof(ChangeNotificationsViewModel), typeof(ChangeNotificationsView), typeof(ChangeNotificationsViewModel_RuntimeGeneratedCode) }),
            };
        }
    }
}
